exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java

```diff
@@ -289,7 +283,13 @@ private IterOutcome sniffNonEmptyBatch(int inputIndex, RecordBatch recordBatch)
           if (recordBatch.getRecordCount() == 0) {
             continue;
           }
-          // We got a non empty batch
+          // We got a non empty batch; update the memory manager
+          final boolean isBuildSide = inputIndex == 1;
+          final int side = isBuildSide ? RIGHT_INDEX : LEFT_INDEX;
```

Comment:
isn't 0 / 1 and LEFT_INDEX / RIGHT_INDEX the same thing? Similarly isn't **side** the same thing as **inputIndex**? Could we make things consistent to avoid confusion? Or if there is a good reason for using different names for things could you add a comment explaining the differences.